article.md

```diff
@@ -0,0 +1,333 @@
+There is a whole dedicated [piece](https://blog.swmansion.com/testing-starknet-contracts-made-easy-with-protostar-2ecdad3c9133) explaining almost everything you would need to know before reading this article, so make sure to check it out. Also, some knowledge about [EIP-712](https://eips.ethereum.org/EIPS/eip-712) would be nice too, as I won't be going into details about how it works.
+
+# Use case - creating a binding between ETH address and Starknet Address
+
+## Message structure
+```json
+{
+  "domain": {
+    "name": "TestContract",
+    "contractAddress": "0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF",
+    "version": "1"
+  },
+
+  "primaryType": "Payload",
+  "types": {
+    "domain": [
+      { "name": "name", "type": "string" },
+      { "name": "contractAddress", "type": "address"},
+      { "name": "version", "type": "string" }
+    ],
+    "Payload": [
+      { "name": "myStarkNetAddress", "type": "uint256" }
+    ]
+  }
+}
+```
+This is the structure that we'll be using in this example. Note that `domain` here represents `eip712Domain` which, after hashing, will become a domain separator that we will need. It ensures that the signature can only be used by our contract on the given chain, and that's what `contractAddress` is needed for (more on that later).<br/>
+`Payload` will be the main structure that we'll be constructing and signing. For the sake of simplicity, it only has one field but the EIP-712 standard allows for any type of structure.
+
+## Creating a connection
+On the surface the concept might be simple, we get the Ethereum address as an input and use the caller's address as Starknet address, we call `save_connected_addresses` function and we're done, right? Well not really, first we have to make sure that the Ethereum address belongs to the person calling this function. That's the hard part, we need to recreate the message signed on the Ethereum side, and then by using the signature given as an input we need to confirm that the extracted address matches our input. That what `assert_valid_eth_signature` function is doing. By calling `get_caller_address()` and using it as our Starknet address we ensure that it indeed belongs to the user and we will use it when reconstructing a signed message.
+```cairo
+@external
+func add_connection{
+    syscall_ptr : felt*,
+    pedersen_ptr : HashBuiltin*,
+    range_check_ptr,
+    bitwise_ptr : BitwiseBuiltin*
+}(eth_address : felt, starknet_address : felt, domain_hash: Uint256, signature_len: felt, signature: felt*){
+    assert_valid_eth_signature(eth_address, starknet_address, domain_hash, signature_len, signature);
+    save_connected_addresses(eth_address, starknet_address);
+    return ();
+}
+```
+
+The hardest part for us is getting the hash of our message. Once we have that, we can use a built-in function `verify_eth_signature_uint256` that will do all the heavy lifting, dealing with complicated math. 
+So we will have to write the `get_hash` function for ourselves.
```

Comment:
You introduce a name of a function before it is defined. I suggest replacing the name with description of what has to be done, like "write a function that constructs an eip712 structure and returns its hash".